### ISSUE TITLE
Parental controls to disable some functionality for young children

### ISSUE THREAD

**Issue by bluetarpmedia** [TortugaPower/BookPlayer#907]
## Feature request

Add a new "protected mode" which would disable certain functionality for young children.

I'd like to hide or disable these features:
- Add files
- Delete files

And whenever the Settings link is clicked it would require a pin code to proceed.

## Narrative

I'd like my young children to be able to listen to audiobooks with BookPlayer on an iPad. But I want to prevent them from accidentally or deliberately trying to import or delete books.

In addition, I'd like the Settings screen to only be accessible via a pin code.

## Design

Add a new toggle to the Settings screen for "Protected / Child Mode". If toggled on, the user must set a pin code.

When Protected Mode is enabled:

All "Add" / "Add files" links are hidden / disabled.
All "Delete" links are hidden / disabled.
Clicking the Settings link requires the pin code to get access to the Settings screen.

**Comment by GianniCarlo** [comment]
This is a very interesting feature request 🤔, definitely something I want to look into, not sure about blocking the entire settings screen with pin code as there are player controls preferences there, but definitely block the tip jar (or the support section which links to Github and mail)

**Comment by bluetarpmedia** [comment]
> not sure about blocking the entire settings screen with pin code as there are player controls preferences there

Yeah, good point. There is the 'Manage Storage' link inside Settings which lets the user delete files, so I was thinking of that plus the external links and tip jar. Maybe enabling 'Protected Mode' could have a subtle link to the tip jar. ;)
